draft-ietf-cose-hash-envelope.md

```diff
@@ -128,31 +129,41 @@ Profiles that rely on this specification MAY choose to mark TBD_1, TBD_2, TBD_3
-      TBD_2: 51                    / "application/json-patch+json" /
-      TBD_3: "https://blob.example/a24f9c19"/ payload_location     /
-    }>>
-    {}                                 / Unprotected               /
-    h'935b5a91...e18a588a',            / Payload                   /
-    h'15280897...93ef39e5'             / Signature                 /
-   ]
-)
+18([ # cose-sign1
+  <<{
+    / signature algorithm / 1: -35, # ES384
+    / key identifier      / 4: h'75726e3a...32636573',
+    / cose sign1 type     / 16: "application/example+cose",
+    / hash algorithm      / TBD_1: -16, # sha256
+    / media type          / TBD_2: "application/spdx+json",
+    / location            / TBD_3: "https://sbom.example/" + ... + "/manifest.spdx.json"
+  }>>
+  / unprotected / {},
+  / payload     / h'935b5a91...e18a588a', # As seen in manifest.spdx.json.sha256
+  / signature   / h'15280897...93ef39e5'  # ECDSA Signature with SHA 384 and P-384
+])
 ~~~~
 
-In this example, the sha256 hash algorithm (-16) is used to hash the payload, which is of content type `application/json-patch+json` identified by the content format `51`.
-The full payload is located at "https://storage.example/244f...9c19".
-The COSE_sign1 is of type "application/example+cose".
-The sha256 hash is signed with ES384 which starts by taking the sha384 hash of the payload (which is a sha256 hash).
+In this example, an SPDX software bill of materials (SBOM) in JSON format is already commonly identified by its SHA256 hash. For example, some tooling generates a file, such as `manifest.spdx.json.sha256`, which contains the SHA256 hash of the corresponding `manifest.spdx.json` file.
+
+The content type for `manifest.spdx.json` is already well known as `application/spdx+json`, and is registered with IANA [here](https://www.iana.org/assignments/media-types/application/spdx+json).
+
+The full JSON SBOM is available at a URL, such as `https://sbom.example/.../manifest.spdx.json`.
+
+The payload of this cose-sign1 is the SHA256 hash of the `manifest.spdx.json`, which is sometimes found in an adjacent file called `manifest.spdx.json.sha256`.
+
+The type of this cose-sign1 is `application/example+cose`, but other types may be used to establish more specific media types for signatures of hashes.
+
+The signature is produced using ES384 which means using ECDSA with SHA384 hash function and P-384 elliptic curve.
+
+This example is chosen to highlight that an existing system may use a hash algorithm such as sha256.
+This hash becomes the payload of a cose-sign1.
+When signed with a signature algorithm that is paramaterized by hash function, such as ECDSA with SHA384, the to be signed structure as described in Section 4.4 of RFC9052.
+
+The resulting signature is over the protected header and payload, providing integrity and authenticity for the hash algorithm, content type and location of the associated resource, in this case a software bill of materials.
```

Comment:
```suggestion
The resulting signature is computed over the protected header and payload, providing integrity and authenticity for the hash algorithm, content type and location of the associated resource, in this case a software bill of materials.
```

Isn't location allowed to be in the unprotected header now?

Reply:
> Isn't location allowed to be in the unprotected header now? 
👎 

However, the updated text doesn't reflect one way or another.